x/slashing/handler.go

```diff
@@ -25,14 +25,23 @@ func handleMsgUnjail(ctx sdk.Context, msg MsgUnjail, k Keeper) sdk.Result {
 		return ErrNoValidatorForAddress(k.codespace).Result()
 	}
 
-	// cannot be unjailed if no self-delegation exists
 	selfDel := k.validatorSet.Delegation(ctx, sdk.AccAddress(msg.ValidatorAddr), msg.ValidatorAddr)
-	if selfDel == nil {
-		return ErrMissingSelfDelegation(k.codespace).Result()
-	}
 
-	if validator.GetDelegatorShareExRate().Mul(selfDel.GetShares()).TruncateInt().LT(validator.GetMinSelfDelegation()) {
-		return ErrSelfDelegationTooLowToUnjail(k.codespace).Result()
+	// A validator attempting to unjail may only do so if its self-bond amount
+	// is at least their declared min self-delegation. However, during disabled
+	// transfers, we allow validators to unjail if they have no self-delegation.
+	// This is to allow newly created validators during a time when transfers are
+	// disabled to successfully unjail.
+	if k.bk.GetSendEnabled(ctx) {
+		if selfDel == nil {
+			// cannot be unjailed if no self-delegation exists
+			return ErrMissingSelfDelegation(k.codespace).Result()
+		}
+
+		valSelfBond := validator.GetDelegatorShareExRate().Mul(selfDel.GetShares()).TruncateInt()
+		if valSelfBond.LT(validator.GetMinSelfDelegation()) {
```

Comment:
You know what I actually think maybe it would beneficial to keep this check in here if validators have increased their self-delegation amount > 1 aka, pre-transfers, if a validator was created _not_ out of an "onbehalfof" transaction they can still add this extra layer of security to their delegators

we could remove this check and add the following check after these embed if statements (function level indentation)
```
if k.bk.GetSendEnabled(ctx) || (!k.bk.GetSendEnabled(ctx) && validator.GetMinSelfDelegation().GT(sdk.OneDec())) {
    if valSelfBond.LT(validator.GetMinSelfDelegation()) {
        return ErrSelfDelegationTooLowToUnjail(k.codespace).Result()
    }
}
```